### ISSUE TITLE
Cpu usage with GGML

### ISSUE THREAD

**Issue by danieljose** [postgresml/postgresml#800]
Hello. This seems to use only 4 cpu cores, but the pc has 16. How to make it use all the cores?
I run it in WSL, but I have not limited it, and in fact to be sure I raised it to 16 CPU cores.
Thanks

```

SELECT pgml.transform(
    task => '{
      "task": "text-generation",
      "model": "TheBloke/falcon-40b-instruct-GGML"
    }'::JSONB,
    inputs => ARRAY[
        'Please summarize the following article:
		June 30 (Reuters) - Artificial intelligence chips...'
    ],
    args => '{"max_new_tokens": 2000}'::JSONB
);

```

**Comment by montanalow** [comment]
Do you have experience with this model using GGML outside of PostgresML, where the parallelization shows different behavior? This thread seems to indicate that this behavior is expected, but I'm not familiar with this particular combination. https://github.com/ggerganov/llama.cpp/issues/578

**Comment by danieljose** [comment]
> Do you have experience with this model using GGML outside of PostgresML, where the parallelization shows different behavior? This thread seems to indicate that this behavior is expected, but I'm not familiar with this particular combination. [ggerganov/llama.cpp#578](https://github.com/ggerganov/llama.cpp/issues/578)

Really I have no experience on this. I understand it's by design, not easy to parallelize this task. Thanks.

**Comment by montanalow** [comment]
If you’re doing more than one inference, you could always use multiple
Postgres connections or background workers to do them in parallel since you
have spare cores, but it’s also nice to have maximum utilization on a
single inference to minimize latency for interactive applications.

If you find anything we could do to improve the underlying dependency, feel
free to reopen.
